### PR TITLE
Iosxe/show ap config general

### PR DESCRIPTION
**Pull request by reustaquio22**
Parser for Iosxe/show ap config general

### CODE REVIEW

File: changelog/undistributed.rst [hunk 1]
```diff
@@ -0,0 +1,19 @@
+* Please follow the template we introduced in NOVEMBER.md file.
```
Comment (LenzDong) [inline changelog/undistributed.rst:1]: We moved the changelog to this folder ```changelog/undistributed```, can you create a new changelog file under this folder. There is a template.rst you can refer to. The file name can be ```216_XE_ShowApConfigGeneral.rst```

https://github.com/CiscoTestAutomation/genieparser/tree/master/changelog/undistributed

File: src/genie/libs/parser/iosxe/tests/test_show_ap.py [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+                "unencrypted_data_keep_alive": "Enabled",
+                "local_dhcp_server": "Disabled",
+                "traffic_distribution_statistics_capability": "Enabled",
+                "dual_dfs_statistics": "Disabled"
+            }
+        }
+    }
+
+    golden_output1 = {'execute.return_value': '''
+Cisco AP Name   : bg-1-cap1
```
Comment (LenzDong) [inline src/genie/libs/parser/iosxe/tests/test_show_ap.py:133]: Could you please fix the indention

File: src/genie/libs/parser/iosxe/show_ap.py [hunk 3]
```diff
@@ -0,0 +1,1446 @@
...
+        ap_name_key = ''
+
+        for line in out_filter:
+            # Cisco AP Name   : bg-1-cap1
+            if cisco_ap_name_capture.match(line):
+                cisco_ap_name_match = cisco_ap_name_capture.match(line)
+                groups = cisco_ap_name_match.groupdict()
+                cisco_ap_name = groups['cisco_ap_name']
+                ap_name_key = cisco_ap_name
+                if not ap_config_general_dict.get('ap_config_general_info', {}):
```
Comment (LenzDong) [inline src/genie/libs/parser/iosxe/show_ap.py:570]: you can use ```setdefault``` method for dictionary like:
```
ap_config_general_dict.setdefault('ap_config_general_info', {})

vs

if not ap_config_general_dict.get('ap_config_general_info', {}):
    ap_config_general_dict['ap_config_general_info'] = {}

File: src/genie/libs/parser/iosxe/show_ap.py [hunk 4]
```diff
@@ -0,0 +1,1446 @@
...
+            elif rogue_containment_of_standalone_flexconnect_aps_capture.match(line):
+                rogue_containment_of_standalone_flexconnect_aps_match = rogue_containment_of_standalone_flexconnect_aps_capture.match(
+                    line)
+                groups = rogue_containment_of_standalone_flexconnect_aps_match.groupdict()
+                rogue_containment_of_standalone_flexconnect_aps = groups[
+                    'rogue_containment_of_standalone_flexconnect_aps']
+                rogue_containment_of_standalone_flexconnect_aps = change_data_type(
+                    value=rogue_containment_of_standalone_flexconnect_aps)
+                ap_config_general_dict['ap_config_general_info'][ap_name_key].update({
+                                                                                         'rogue_containment_of_standalone_flexconnect_aps': rogue_containment_of_standalone_flexconnect_aps})
```
Comment (LenzDong) [inline src/genie/libs/parser/iosxe/show_ap.py:1216]: please fix indention

File: src/genie/libs/parser/iosxe/tests/test_show_ap.py [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+# ======================================
+# Unit test for 'show ap config general'
+# ======================================
+class TestShowApConfigGeneral(unittest.TestCase):
+    """Unit test for 'show ap config general'"""
+
+    maxDiff = None
+    empty_output = {'execute.return_value': ''}
+    golden_parsed_output1 = {
+        "ap_config_general_info": {
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/tests/test_show_ap.py:17]: better to have key name for the value. so `ap_name` is better?

File: src/genie/libs/parser/iosxe/tests/test_show_ap.py [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+            "bg-1-cap1": {
+                "cisco_ap_identifier": "70b3.d278.e03e",
+                "country_code": "IN",
+                "regulatory_domain_allowed_by_country": "802.11bg:-A   802.11a:-DN",
+                "ap_country_code": "IN  - India",
+                "slot_0": "-A",
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/tests/test_show_ap.py:23]: based on output with indentation, better to have like below:
```
'ap_regulatory_domain': {
  'slot': {
    0: "-A",
    1: "-D",
   }
   (snip)

File: src/genie/libs/parser/iosxe/show_ap.py [hunk 7]
```diff
@@ -0,0 +1,1446 @@
...
+# Schema for:
+#  * 'show ap config general'
+# ===========================
+class ShowApConfigGeneralSchema(MetaParser):
+    """Schema for show ap config general."""
+
+    schema = {
+        "ap_config_general_info": {
+            str: {
+                Optional("cisco_ap_identifier"): str,
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/show_ap.py:17]: all of them are `Optional`? if there are mandatories one, please remote from them.

**Pull request closed without merging**